### PR TITLE
vg: make TestLineWidth more reliable

### PR DESCRIPTION
**Pull request by sbinet**
golang/go@53efe1e121e68fe changed and improved `compress/flate`'s
matching algorithm.
This breaks tests that rely on stable on-disk byte content.

Instead, decode the image.Image from the raw []byte content and compare
the resulting decoding images.

For PDF images, decode the raw []byte using rsc.io/pdf.Reader and
compare each page's content.

### CODE REVIEW

**Comment by sbinet** [comment]
tested on go-tip (`go version devel +1f5b1b2 Mon Apr 4 10:05:05 2016 +0000 linux/amd64`)

**Comment by eaburns** [comment]
Sounds fine. But, why not do the simpler byte-equality on the PDFs too?

**Comment by eaburns** [comment]
Sorry, I read this before the comment on #250. Does the PDF change here fix all of the breakage in the PDF tests that you mentioned in #250?

**Comment by sbinet** [comment]
yes. at golang/go@1f5b1b2, `go get ./... && go test ./...` succeeds flawlessly and completely.

**Comment by eaburns** [comment]
LGTM.

File: vg/vg_test.go [hunk 1]
```diff
@@ -65,22 +68,90 @@ func TestLineWidth(t *testing.T) {
...
+				if err != nil {
+					t.Fatalf("failed to retrieve test image infos: %v", err)
+				}
+				want, err := pdf.NewReader(f, fi.Size())
+				if err != nil {
+					t.Fatalf("failed to decode test image (typ=%s): %v", typ, err)
+				}
+
+				r := bytes.NewReader(buf.Bytes())
+				// FIXME: bytes.Reader.Size was introduced only after go-1.4
```
Comment (kortschak) [inline vg/vg_test.go]: Make this a TODO(sbinet) rather than a FIXME.

**Comment by kortschak** [comment]
Nice. LGTM after minor change.

File: vg/vg_test.go (hunk 1 above)
Comment (sbinet) [inline vg/vg_test.go]: done.

**Pull request merged**